orangecontrib/spectroscopy/widgets/owhyper.py

```diff
@@ -812,15 +920,17 @@ def on_done(self, res):
         if invalid_positions:
             self.parent.Information.not_shown(invalid_positions)
 
-        imdata = np.ones((lsy[2], lsx[2])) * float("nan")
+        imdata = np.ones((lsy[2], lsx[2], d.shape[1])) * float("nan")
         imdata[yindex[valid], xindex[valid]] = d[valid]
+
         self.data_values = d
         self.data_imagepixels = np.vstack((yindex, xindex)).T
-
         self.img.setImage(imdata, autoLevels=False)
         self.update_levels()
+        self.update_rgb_levels()
         self.update_color_schema()
         self.update_legend_visible()
+        self.update_levels_menu_visible()
```

Comment:
Levels menu visibility should be set when the corresponding UI choices were made, not after on_done.

Reply:
Okay